scico/linop/radon_svmbir.py

```diff
@@ -227,18 +234,35 @@ def __init__(
 
         self.has_prox = True
 
-        if prox_kwargs is None:
-            prox_kwargs = dict()
+        default_prox_args = {"maxiter": 1000, "ctol": 0.001}
+        default_prox_args.update(prox_kwargs)
 
         svmbir_prox_args = dict()
-        if "maxiter" in prox_kwargs:
-            svmbir_prox_args["max_iterations"] = prox_kwargs["maxiter"]
-        if "ctol" in prox_kwargs:
-            svmbir_prox_args["stop_threshold"] = prox_kwargs["ctol"]
+        if "maxiter" in default_prox_args:
+            svmbir_prox_args["max_iterations"] = default_prox_args["maxiter"]
+        if "ctol" in default_prox_args:
+            svmbir_prox_args["stop_threshold"] = default_prox_args["ctol"]
         self.svmbir_prox_args = svmbir_prox_args
 
         self.positivity = positivity
 
+        if W is None:
+            self.W = Identity(self.y.shape)
+        elif isinstance(W, Diagonal):
+            if snp.all(W.diagonal >= 0):
+                self.W = W
+            else:
+                raise Exception(f"The weights, W, must be non-negative.")
```

Comment:
This is checked by the `__init__` of `WeightedSquaredL2Loss`. Would it be possible to reduce duplication by making use of `super().__init__`?

Reply:
The `SVMBIRExtendedLoss` inherits from `Loss` instead of `WeightedSquaredL2Loss` since it is not a true WeightedSquaredL2Loss. So `super().__init__` won't work here since `Loss` doesn't have the weight.

One Way to deduce duplication would be to define a `WeightedLoss ` that inherits from loss. Both `WeightedSquaredL2Loss` and `SVMBIRExtendedLoss` can inherit from this.